mender-update/deployments/platform/boost_log/deployments.cpp

```diff
@@ -0,0 +1,197 @@
+	fs::path dir_path(data_store_dir_);
+	bool created = fs::create_directories(dir_path);
+	if (created) {
+		// should basically never happen, but if we happened to create the
+		// directory, it's empty and thus well-prepared
+		return error::NoError;
+	}
+
+	vector<string> old_logs;
+	fs::directory_iterator end_itr; // default construction yields past-the-end
+	for (fs::directory_iterator itr(dir_path); itr != end_itr; ++itr) {
+		fs::path file_path = itr->path();
+		if (!fs::is_regular_file(file_path)) {
+			continue;
+		}
+
+		string file_name = file_path.filename().string();
+
+		if (file_name == "deployments.0000." + id_ + ".log") {
+			// this log file will be (re)used, leave it alone
+			continue;
+		}
+
+		if ((file_name.find("deployments.") != 0)
+			|| (file_name.substr(file_name.size() - 4) != ".log")) {
+			continue;
+		}
+
+		old_logs.push_back(file_name);
+	}
+	std::sort(old_logs.begin(), old_logs.end());
+
+	error_code ec;
+	fs::space_info space_info = fs::space(dir_path, ec);
+	if (ec) {
+		return error::Error(
+			ec.default_error_condition(), "Failed to check free space for log files");
+	}
+
+	while ((old_logs.size() > 0)
+		   && ((space_info.available < kLogsFreeSpaceRequired)
+			   || (old_logs.size() > (kMaxExistingLogs - 1)))) {
+		auto last_log_file = old_logs[old_logs.size() - 1];
+		old_logs.pop_back();
+		if (!fs::remove(dir_path / last_log_file, ec) && ec) {
+			return error::Error(
+				ec.default_error_condition(),
+				"Failed to remove old log file '" + last_log_file + "'");
+		}
+		if (space_info.available < kLogsFreeSpaceRequired) {
+			space_info = fs::space(dir_path, ec);
+			if (ec) {
+				return error::Error(
+					ec.default_error_condition(), "Failed to check free space for log files");
+			}
+		}
+	}
+
+	// now let's make sure old logs have an increasing index starting with 0001
+	for (ssize_t i = old_logs.size() - 1; i >= 0; i--) {
+		// "deployments.".size() == 12
+		auto second_dot_pos = old_logs[i].find('.', 12);
+		auto last_dot_pos = old_logs[i].find_last_of('.');
+
+		// should never happen due the filter above when populating old_logs and
+		// due to how these files are named
+		assert(second_dot_pos != string::npos);
+		assert(last_dot_pos != string::npos);
+
+		string deployment_id;
+		if ((second_dot_pos == string::npos) || (last_dot_pos == string::npos)) {
+			deployment_id = "unknown_deployment";
+		} else {
+			deployment_id =
+				old_logs[i].substr(second_dot_pos + 1, (last_dot_pos - second_dot_pos - 1));
+		}
+		stringstream ss;
+		ss << "deployments.";
+		ss << setfill('0') << setw(4) << to_string(i + 1);
+		ss << "." + deployment_id;
+		ss << ".log";
+
+		string new_name = ss.str();
+		fs::rename(dir_path / old_logs[i], dir_path / new_name, ec);
+		if (ec) {
+			return error::Error(
+				ec.default_error_condition(),
+				"Failed to rename old log file '" + old_logs[i] + "'");
+		}
+	}
+
+	return error::NoError;
+}
+
+error::Error DeploymentLog::BeginLogging() {
```

Comment:
S: 

Why not call it just `Begin`. the variable is already something deployment logger, right?

Reply:
It's called `DeploymentLog` and it actually begins logging when you call this function. So to me it makes sense to call it `BeginLogging()`.

Reply:
Yeh, just a suggestion.

But doesn't:

`deployment_logger.Begin()` 

make sense?